### ISSUE TITLE
Support ignore lists or include/exclude patterns

### ISSUE THREAD

**Issue by fwextensions** [loreanvictor/tmplr#32]
There are likely to be a number of files in a repo that are used only for scaffolding tmplr, so it would be nice if there was a simple way of not copying them to the output directory.  It's possible to use `remove`, but it somehow seems cleaner to not copy them in the first place.

**Comment by fwextensions** [comment]
Maybe anything with `tmplr` in the filename could be ignored automatically, so a `package.tmplr.json` file could be used as a source for copying to an output file, but it would never be copied on its own.

**Comment by loreanvictor** [comment]
Wouldn't `remove: **/*.tmplr.*` achieve the same thing? In any case, such files aren't particularly _copied_ over, they are cloned with the rest of the files and would need to be removed in the end.

Another solution, which is a bit more involved, would be to use [temp directories](https://github.com/loreanvictor/tmplr#temporary-directories). You could clone whatever files into a temp directory and only copy what you need from it. The whole directory would then be removed after the recipe has finished executing.

```yaml
steps:
  - degit: user/repo
    to:
      eval: '{{ tmpdir.repo }}'
      
  ...
```

**Comment by lintong** [comment]
Semi related issue but, I've found that it's overwriting my `.git` folder when applying this using a `local:` template. This wouldn't happen if `.git` could be explicitly or implicitly ignored

**Comment by loreanvictor** [comment]
yeah I think this is a bug, specifically [here in @tmplr/node](https://github.com/loreanvictor/tmplr-node/blob/60ec9f2fad8e33d5445c51f0cd4846824df360be/src/filesystem/node-fs.ts#L113-L124):

```js
    if (remote.startsWith('local:')) {
      const path = this.absolute(remote.slice(6))
      await cp(path, abs, { recursive: true })
    } else {
      const emitter = tiged(remote, {
        subgroup: options?.subgroup ?? false,
        cache: false,
        force: true
      })

      await emitter.clone(abs)
    }
```

basically, the code will copy the target folder if it is local, but clone it if it is a repo using tiged, in the latter case `.git` being skipped and not overriden, but not in the former case. I will try to reproduce this, and then create the bug report on [@tmplr/node](https://github.com/loreanvictor/tmplr-node).

p.s. @lintong this might also be a good first issue if you want to take a shot at it 😉
